src/sage/crypto/sboxes.py

```diff
@@ -166,6 +166,9 @@
 """
 import sys
 
+from sage.rings.integer_ring import ZZ
+from sage.rings.finite_rings.finite_field_constructor import GF
+from sage.modules.free_module_element import vector
 from sage.crypto.sbox import SBox
```

Comment:
If you see lines 222,223 and 396,397 in this file, imports are placed in each function using them. I think you should follow the style.  This style may be related with https://doc-release--sagemath.netlify.app/html/en/developer/packaging_sage_library#module-level-runtime-dependencies though I am not sure.